src/FileIO.h

```diff
@@ -60,4 +61,26 @@ struct ScopedFileDescriptor {
       close(fd);
    }
 };
+
+// Do file access as root user for the function passed in.
+// Example: auto result = FileIO::SudoFile(FileIO::ReadAsciiFileContent, filePath);
+template<typename FnCall, typename... Args>
+auto SudoFile(FnCall fn, Args&& ... args) {
```

Comment:
i.e. it *also* shows that the template substitution failed but it's way too much information. Unless you are skilled with templates the first error message is much easier to understand.